app/main.py

```diff
@@ -1,2 +1,55 @@
+from typing import Any
+
+
 class Dictionary:
-    pass
+    def __init__(self) -> None:
+        self.capacity = 8
+        self.size = 0
+        self.threshold = int(self.capacity * 2 / 3)
+        self.hash_table = [None for _ in range(self.capacity)]
+
+    def __setitem__(self, key: Any, value: Any) -> None:
+        if self.size == self.threshold:
+            self.resize()
+        hash_ = hash(key)
+        index_ = hash_ % self.capacity
+        if self.hash_table[index_] is None:
+            self.hash_table[index_] = (key, hash_, value)
+            self.size += 1
+            return
+        if self.hash_table[index_][0] == key:
+            self.hash_table[index_] = (key, hash_, value)
+            return
+        for _ in range(self.capacity):
+            index_ = (index_ + 1) % self.capacity
+            if self.hash_table[index_] is None:
+                self.hash_table[index_] = (key, hash_, value)
+                self.size += 1
+                return
+            if self.hash_table[index_][0] == key:
+                self.hash_table[index_] = (key, hash_, value)
+                return
+
+    def __getitem__(self, key: Any) -> Any:
+        hash_ = hash(key)
+        index_ = hash_ % self.capacity
+        for _ in range(self.capacity):
```

Comment:
You may rewrite this loop as follows:
```python
while self.hash_table[index] is not None:
```
This will allow you to remove the first check.